lib/scope.js

```diff
@@ -103,11 +103,15 @@ AST_Toplevel.DEFMETHOD("figure_out_scope", function(options){
     var nesting = 0;
     var in_destructuring = null;
     var tw = new TreeWalker(function(node, descend){
-        if (options.screw_ie8 && node instanceof AST_Catch) {
+        var create_a_block_scope =
```

Comment:
Cool, making it

```
        var create_a_block_scope =
            (options.screw_ie8 && node instanceof AST_Catch) ||
            ((node instanceof AST_Block) && node.creates_block_scope())
```

There may be some nuances as to how ES6 modules behave in terms of the toplevel scope, so this might be useful for that too.